tests/sentry/tasks/test_post_process.py

```diff
@@ -994,9 +897,184 @@ def test_ensure_when_assignees_and_owners_are_cached_does_not_cause_unbound_erro
+
+    @patch("sentry.rules.processor.RuleProcessor")
+    def test_maintains_valid_snooze(self, mock_processor):
+        event = self.create_event(data={}, project_id=self.project.id)
+        cache_key = write_event_to_cache(event)
+        group = event.group
+        snooze = GroupSnooze.objects.create(group=group, until=timezone.now() + timedelta(hours=1))
+
+        self.call_post_process_group(
+            is_new=True,
+            is_regression=False,
+            is_new_group_environment=True,
+            cache_key=cache_key,
+            group_id=event.group_id,
+        )
+
+        mock_processor.assert_called_with(EventMatcher(event), True, False, True, False)
+
+        assert GroupSnooze.objects.filter(id=snooze.id).exists()
+
+
 @region_silo_test
-class PostProcessGroupPerformanceTest(TestCase, SnubaTestCase, PerfIssueTransactionTestMixin):
-    @with_feature("organizations:performance-issues-post-process-group")
+class PostProcessGroupErrorTest(
+    TestCase,
+    AssignmentTestMixin,
+    CorePostProcessGroupTestMixin,
+    InboxTestMixin,
+    ResourceChangeBoundsTestMixin,
+    RuleProcessorTestMixin,
+    ServiceHooksTestMixin,
+    SnoozeTestMixin,
+):
+    def create_event(self, data, project_id):
+        return self.store_event(data=data, project_id=project_id)
+
+    def call_post_process_group(
+        self, is_new, is_regression, is_new_group_environment, cache_key, group_id
+    ):
+        post_process_group(
+            is_new=is_new,
+            is_regression=is_regression,
+            is_new_group_environment=is_new_group_environment,
+            cache_key=cache_key,
+            group_id=group_id,
+        )
+
+
+@region_silo_test
+@apply_feature_flag_on_cls("organizations:performance-issues-post-process-group")
+class PostProcessGroupPerformanceTest(
+    TestCase,
+    SnubaTestCase,
+    PerfIssueTransactionTestMixin,
+    CorePostProcessGroupTestMixin,
+    InboxTestMixin,
+    RuleProcessorTestMixin,
+    SnoozeTestMixin,
+):
+    def create_event(self, data, project_id):
+        fingerprint = data["fingerprint"][0] if data.get("fingerprint") else "some_group"
+        fingerprint = f"{GroupType.PERFORMANCE_N_PLUS_ONE.value}-{fingerprint}"
+        # Store a performance event
+        event = self.store_transaction(
+            project_id=project_id,
+            user_id="hi",
+            fingerprint=[fingerprint],
+        )
+        return event.for_group(event.groups[0])
+
+    def call_post_process_group(
+        self, is_new, is_regression, is_new_group_environment, cache_key, group_id
+    ):
+        group_states = (
+            [
+                {
+                    "id": group_id,
+                    "is_new": is_new,
+                    "is_regression": is_regression,
+                    "is_new_group_environment": is_new_group_environment,
+                }
+            ]
+            if group_id
+            else None
+        )
```

Comment:
Hmm, any reason not to add `group_states` as an optional arg on `call_post_process_group`.

The test `PostProcessGroupPerformanceTest.test_full_pipeline_with_group_states` passes in two GroupState objects but this won't support that.

Reply:
I suppose we could allow `group_states` to be passed as an override. tbh I was just calling it directly from that test since it's a special case